core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -116,6 +116,7 @@ public class RESTSessionCatalog extends BaseViewSessionCatalog
   private static final String DEFAULT_FILE_IO_IMPL = "org.apache.iceberg.io.ResolvingFileIO";
   private static final String REST_METRICS_REPORTING_ENABLED = "rest-metrics-reporting-enabled";
   private static final String REST_SNAPSHOT_LOADING_MODE = "snapshot-loading-mode";
+  static final String REST_NAMESPACE_SEPARATOR = "rest-namespace-separator";
```

Comment:
There's no need for `rest-` here and we should always try to make config properties as short as possible while being clear.

Reply:
I was trying to align with the other config properties, but renamed it to `namespace-separator`